src/binlog.cpp

```diff
@@ -77,25 +77,23 @@ void BinLog::_send_stop()
 int BinLog::write_msg(const struct buffer *buffer)
 {
     const bool mavlink2 = buffer->data[0] == MAVLINK_STX;
-    uint32_t msg_id;
-    uint8_t *payload;
-    uint16_t payload_len;
-    uint8_t trimmed_zeros;
-    uint8_t source_system_id;
-    uint8_t source_component_id;
-    mavlink_remote_log_data_block_t *binlog_data;
+    uint32_t msg_id = 0;
+    uint8_t *payload = nullptr;
+    uint16_t payload_len = 0;
+    uint8_t trimmed_zeros = 0;
+    uint8_t source_system_id = 0;
+    uint8_t source_component_id = 0;
+    mavlink_remote_log_data_block_t *binlog_data = nullptr;
```

Comment:
as I said previously, leaving it unsigned is a feature - all of those are supposed to be assigned in one of the branches below. If we initialize them, then we basically lose the convenience of running valgrind to detect bugs.

Reply:
Sounds a bit strange to me. But I can leave that check out, if you like it better the way it was.

Reply:
This recommendation is pretty bad. It came from Java or maybe from when gcc/clang wouldn't be able to help you. Nowadays compilers have -Wuninitialized / -Wmaybe-uninitialized and valgrind also helps you. If you force the initialization of local variables just for the sake of it, then it makes it harder for you to root cause bugs